pkg/operation/botanist/controlplane/konnectivity/konnectivity_server.go

```diff
@@ -0,0 +1,552 @@
+			Name:      name,
+			Namespace: k.Namespace,
+			Labels:    getLabels(),
+		}}
+		deployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: k.Namespace,
+				Labels:    getLabels(),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas:             pointer.Int32Ptr(k.Replicas),
+				RevisionHistoryLimit: pointer.Int32Ptr(0),
+				Selector:             &metav1.LabelSelector{MatchLabels: getLabels()},
+				Strategy: appsv1.DeploymentStrategy{
+					Type: appsv1.RollingUpdateDeploymentStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateDeployment{
+						MaxUnavailable: &intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+						MaxSurge:       &intstr.IntOrString{Type: intstr.Int, IntVal: 1},
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"checksum/secret-" + k.secrets.Kubeconfig.Name: k.secrets.Kubeconfig.Checksum,
+							"checksum/secret-" + k.secrets.Server.Name:     k.secrets.Server.Checksum,
+							"checksum/secret-" + k.secrets.ClientCA.Name:   k.secrets.ClientCA.Checksum,
+						},
+						Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+							v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyFromShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToShootAPIServer:   v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToSeedAPIServer:    v1beta1constants.LabelNetworkPolicyAllowed,
+						}),
+					},
+					Spec: corev1.PodSpec{
+						Affinity: &corev1.Affinity{
+							PodAntiAffinity: &corev1.PodAntiAffinity{
+								PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+									Weight: 100,
+									PodAffinityTerm: corev1.PodAffinityTerm{
+										TopologyKey:   corev1.LabelHostname,
+										LabelSelector: &metav1.LabelSelector{MatchLabels: getLabels()},
+									},
+								}},
+							},
+						},
+						ServiceAccountName: name,
+						Containers: []corev1.Container{{
+							Name:            name,
+							Image:           k.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								fmt.Sprintf("--namespace=%s", k.Namespace),
+								fmt.Sprintf("--deployment-name=%s", name),
+								"--jitter=10s",
+								"--jitter-factor=5",
+								"--v=2",
+								"--",
+								"/proxy-server",
+								"--logtostderr=true",
+								"--cluster-cert=/certs/konnectivity-server/konnectivity-server.crt",
+								"--cluster-key=/certs/konnectivity-server/konnectivity-server.key",
+								"--server-cert=/certs/konnectivity-server/konnectivity-server.crt",
+								"--server-key=/certs/konnectivity-server/konnectivity-server.key",
+								"--server-ca-cert=/certs/client-ca/ca.crt",
```

Comment:
Can you have some variables for these paths that can be used both here and below in the `volumeMount` section?